python/GangaLHCb/Lib/Applications/GaudiExec.py

```diff
@@ -204,11 +204,11 @@ def prepare(self, force=False):
         # file is unspecified, has a space or is a relative path
         self.configure(self)
         logger.info('Preparing %s application.' % getName(self))
-        self.is_prepared = ShareDir()
-        logger.info('Created shared directory: %s' % (self.is_prepared.name))
 
         this_build_target = self.buildGangaTarget()
 
+        self.is_prepared = ShareDir()
+        logger.info('Created shared directory: %s' % (self.is_prepared.name))
```

Comment:
Should the lines not be moved to the end of the try clause just below - so that we only declare that it is prepared when the files are actually copied into their final destination.